### PR TITLE
Add multimon support

### PR DESCRIPTION
**Pull request by kubistika**
This PR adds multimon support.

### CODE REVIEW

File: server/proxy/pf_server.c [hunk 1]
```diff
@@ -155,12 +155,18 @@ static BOOL pf_server_post_connect(freerdp_peer* client)
 	}
 
 	pc = (pClientContext*) p_client_context_create(client->settings, host, port);
-	connectionClosedEvent = CreateEvent(NULL, TRUE, FALSE, NULL);
+
+	if (pc == NULL)
```
Comment (kubistika) [inline server/proxy/pf_server.c]: should be removed from this PR

File: server/proxy/pf_server.c [hunk 2]
```diff
@@ -155,12 +155,18 @@ static BOOL pf_server_post_connect(freerdp_peer* client)
...
+	{
+		WLog_ERR(TAG, "p_client_context_create failed!");
+		return FALSE;
+	}
+
 	/* keep both sides of the connection in pdata */
 	pc->pdata = ps->pdata;
 	pdata->pc = (pClientContext*) pc;
 	pdata->ps = ps;
-	pdata->connectionClosed = connectionClosedEvent;
```
Comment (kubistika) [inline server/proxy/pf_server.c]: should be removed from this PR

File: libfreerdp/core/display.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#include "display.h"
```
Comment (speidy) [inline libfreerdp/core/display.c:20]: add copyright

File: libfreerdp/core/display.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		Stream_Write_UINT32(s, monitor->flags); /* flags (4 bytes) */
+	}
+
+	return TRUE;
+}
+
+void display_convert_rdp_monitor_to_monitor_def(UINT32 monitorCount,
+        const rdpMonitor* monitorDefArray, MONITOR_DEF** result)
+{
+	UINT32 index;
```
Comment (speidy) [inline libfreerdp/core/display.c:48]: check args

File: server/proxy/pf_client.c [hunk 5]
```diff
@@ -127,6 +137,7 @@ static BOOL pf_client_pre_connect(freerdp* instance)
 		return FALSE;
 	}
 
+	instance->update->RemoteMonitors = pf_client_remote_monitors;
```
Comment (speidy) [inline server/proxy/pf_client.c]: maybe its better to put it at the top, after `settings->` stuff

File: server/proxy/pf_context.c [hunk 6]
```diff
@@ -71,11 +71,52 @@ rdpContext* p_client_context_create(rdpSettings* clientSettings,
 	settings = context->settings;
 	pf_common_copy_settings(settings, clientSettings);
 	settings->Username = _strdup(clientSettings->Username);
-	settings->Password = _strdup(clientSettings->Password);
+	settings->Password = _strdup("123123");
```
Comment (speidy) [inline server/proxy/pf_context.c]: no

File: server/proxy/pf_context.c [hunk 7]
```diff
@@ -71,11 +71,52 @@ rdpContext* p_client_context_create(rdpSettings* clientSettings,
...
 	settings->Domain = _strdup(clientSettings->Domain);
 	settings->ServerHostname = host;
 	settings->ServerPort = port;
 	settings->SoftwareGdi = FALSE;
 	settings->RedirectClipboard = FALSE;
+	/* Client Monitor Data */
```
Comment (speidy) [inline server/proxy/pf_context.c:80]: space before

File: server/proxy/pf_context.c [hunk 8]
```diff
@@ -71,11 +71,52 @@ rdpContext* p_client_context_create(rdpSettings* clientSettings,
...
+	settings->UseMultimon = clientSettings->UseMultimon;
+	settings->ForceMultimon = clientSettings->ForceMultimon;
+	settings->DesktopPosX = clientSettings->DesktopPosX;
+	settings->DesktopPosY = clientSettings->DesktopPosY;
+	settings->ListMonitors = clientSettings->ListMonitors;
+	settings->NumMonitorIds = clientSettings->NumMonitorIds;
+	settings->MonitorLocalShiftX = clientSettings->MonitorLocalShiftX;
+	settings->MonitorLocalShiftY = clientSettings->MonitorLocalShiftY;
+	settings->HasMonitorAttributes = clientSettings->HasMonitorAttributes;
+	settings->NegotiationFlags = clientSettings->NegotiationFlags;
```
Comment (speidy) [inline server/proxy/pf_context.c]: im not sure about that one

File: server/proxy/pf_context.c (hunk 6 above)
Comment (kubistika) [reply server/proxy/pf_context.c]: lol

File: server/proxy/pf_context.c (hunk 7 above)
Comment (kubistika) [reply server/proxy/pf_context.c:80]: formatter doesn't allow

**Pull request merged**